src/main/java/hudson/plugins/jobConfigHistory/FileHistoryDao.java

```diff
@@ -320,6 +320,38 @@ public void deleteItem(final Item item) {
 		}
 	}
 
+	@Override
+	public void changeItemLocation(Item item, String oldFullName, String newFullName) {
+		//do the old switcheroo
+		final String onLocationChangedDescription = "old full name: " + oldFullName
+				+ ", new full name: " + newFullName;
+		if (historyRootDir != null) {
+			final String jobsStr = "/jobs/";

```

Comment:
This is problematic on non-Unix hosts. There's a [isUnix()](https://javadoc.jenkins.io/jenkins/model/Jenkins.MasterComputer.html#isUnix--) method that you could use to check if you're running on a Unix-like host. But I'm not sure if that's the best way here. You might check the [Folders plugin](https://github.com/jenkinsci/cloudbees-folder-plugin) for methods that could be used to generate directories directly.

Reply:
I used a unix check because the Jenkins API does only provide the old item's "full name", not an item object so it wasn't possible to use preexistant methods for the old history directory.